force-app/main/default/classes/SOQL.cls

```diff
@@ -314,28 +434,51 @@ public with sharing class SOQL {
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
+        List<OpportunityLineItem> oppLineItems = [
+            SELECT Id, OpportunityId, Product2Id, Quantity, UnitPrice, PricebookEntryId
+            FROM OpportunityLineItem
+            WHERE OpportunityId = :origOppId
+        ];
 
         // Create a list to hold the new line items
+        List<OpportunityLineItem> clonedLineItems = new List<OpportunityLineItem>();
 
         // Generate a random number between 1 and 3
+        Integer random1To3 = (Integer)Math.ceil(Math.random() * 3);
 
         // Loop through each cloned Opportunity Id
-
-            // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
+        for (Id cloneOppId : cloneOppIds) {
+            for (OpportunityLineItem origLineItem : oppLineItems) {
+                // Clone the original line item
+                OpportunityLineItem newLineItem = new OpportunityLineItem(
+                    OpportunityId = cloneOppId,
+                    PricebookEntryId = origLineItem.PricebookEntryId,
+                    Quantity = origLineItem.Quantity + random1To3,
+                    UnitPrice = origLineItem.UnitPrice
+                );
+                clonedLineItems.add(newLineItem);
+            }
+        }
```

Comment:
Very nice!
You were able to clone each line item, now try to use the [Clone()](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_sobject.htm#apex_System_SObject_clone) method for practice.